ciongke/cmd/deck/main.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/Sirupsen/logrus"
+
+	"github.com/kubernetes/test-infra/ciongke/kube"
+)
+
+const (
+	namespace = "default"
+)
+
+var ja *JobAgent
+
+func main() {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+
+	kc, err := kube.NewClientInCluster(namespace)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting client.")
+	}
+
+	ja = &JobAgent{
+		kc: kc,
+	}
+	ja.Start()
+
+	http.Handle("/", http.FileServer(http.Dir("/static")))
+	http.HandleFunc("/data.js", func(w http.ResponseWriter, r *http.Request) {
+		jobs := ja.Jobs()
+		jd, err := json.Marshal(jobs)
+		if err != nil {
+			logrus.WithError(err).Error("Error marshaling jobs.")
+			jd = []byte("[]")
+		}
+		w.Header().Set("Cache-Control", "no-cache")
+		fmt.Fprintf(w, "var allBuilds = %s;", string(jd))
```

Comment:
Agreed, but the client script needs to implicitly know that it will create a variable named allBuilds. Client code might even have this variable name already (unlikely, I agree). A common practice with json-p is that the client can request the name of the variable by doing
`src=data.js?varName=allBuilds`, the server will reply `fmt.Sprintf("var %s = %s;", varName, data)`